runtime/src/launch/slurm-gasnetrun_ibv/launch-slurm-gasnetrun_ibv.c

```diff
@@ -160,29 +160,31 @@ static void genNumLocalesOptions(FILE* slurmFile, sbatchVersion sbatch,
 
 static void propagate_environment(FILE *f)
 {
+  // Indiscriminately propagate all environment variables.
+  // We could do this more selectively, but we would be likely
+  // to leave out something important.
+  char *enviro_keys = chpl_get_enviro_keys(',');
+  if (enviro_keys)
+    fprintf(f, " -E %s", enviro_keys);
```

Comment:
FYI this is breaking for me on an internal cray-cs. It's trying to pass `BASH_FUNC_module()` and slurm complains with:

```
/var/spool/slurmd/job25388/slurm_script: line 8: syntax error near unexpected token `('
/var/spool/slurmd/job25388/slurm_script: line 8: `/lus/scratch/eronagha/chapel/third-party/gasnet/install/cray-cs-gnu-unknown-none/substrate-ibv/seg-large/nodbg/bin//gasnetrun_ibv -n 2 -N 2 -E CHPL_LAUNCHER_DEBUG,MODULE_VERSION_STACK,MANPATH,XDG_SESSION_ID,HOSTNAME,TERM,SHELL,SLURM_MPI_TYPE,HISTSIZE,SSH_CLIENT,QTDIR,OLDPWD,QTINC,CHPL_LAUNCHER_USE_SBATCH,SSH_TTY,QT_GRAPHICSSYSTEM_CHECKED,USER,LD_LIBRARY_PATH,LS_COLORS,CHPL_HOST_PLATFORM,PE_PRODUCT_LIST,MODULE_VERSION,MAIL,PATH,GASNET_PHYSMEM_NOPROBE,PWD,_LMFILES_,LANG,MODULEPATH,KDEDIRS,LOADEDMODULES,HISTCONTROL,CHPL_HOME,SHLVL,HOME,CHPL_LAUNCHER,DYLD_LIBRARY_PATH,GASNET_PHYSMEM_MAX,CRAYPE_NETWORK_TARGET,LOGNAME,QTLIB,XDG_DATA_DIRS,SSH_CONNECTION,MODULESHOME,LESSOPEN,CRAY_CPU_TARGET,XDG_RUNTIME_DIR,QT_PLUGIN_PATH,BASH_FUNC_module(),_ env LANG=en_US.UTF-8 LC_ALL= LC_COLLATE= ./hello_real  '-nl2''
```

Is there a user issue this was solving? I see https://github.com/chapel-lang/chapel/issues/11355, but is there a mail or something you can link me to?

Slurm propagates environment variables, so is the issue that env vars might be propagated through the salloc, but still not through the actual launch via `gasnetrun_ibv`?

Reply:
This is fixed in https://github.com/chapel-lang/chapel/pull/12339